### PR TITLE
Isis Pagination  update

### PR DESCRIPTION
**Pull request by brianteeman**
Improve the accessibility of the pagination - there are no visual changes

1. Wrap the pagination in a nav and add a role (needed for IE) and an aria-label
2. Ensure font icons have aria-hidden=true
3. Add aria-current to the selected page
4. Add aria-title to the active links eg Go to page 1

### Note 1
This PR does not include the language strings as they were added in #18326

### Note 2
I dont know why but isis and protostar (see #18326) use an override for pagination instead of the layout. For b/c this PR is just for the override but when approved I will do a similar PR for the layout

Thanks to @fuzzbomb (Drupal 8 Core Accessibility Maintainer) for his advice and reviewing this

### CODE REVIEW

File: administrator/templates/isis/html/layouts/joomla/pagination/link.php [hunk 1]
```diff
@@ -30,34 +30,39 @@
...
+		$aria = JText::sprintf('JLIB_HTML_GOTO_PAGE', $item->text);		
 		break;
 }
 
 $item->text .= $addText ?: '';
 
 if ($icon !== null)
 {
-	$display = '<span class="' . $icon . '"></span>';
+	$display = '<span class="' . $icon . ' aria-hidden="true"></span>';
```
Comment (Quy) [inline administrator/templates/isis/html/layouts/joomla/pagination/link.php]: Add closing `"` to the `class` attribute.

File: administrator/templates/isis/html/layouts/joomla/pagination/link.php [hunk 2]
```diff
@@ -95,12 +100,14 @@
 ?>
 <?php if ($displayData['active']) : ?>
 	<li<?php echo $liClass ? ' class="' . $liClass . '"' : ''; ?>>
-		<a <?php echo $cssClasses ? 'class="' . implode(' ', $cssClasses) . '"' : ''; ?> <?php echo $title; ?> href="#" onclick="<?php echo $onClick; ?>">
+		<a aria-label="<?php echo $aria;?>" <?php echo $cssClasses ? 'class="' . implode(' ', $cssClasses) . '"' : ''; ?> <?php echo $title; ?> href="#" onclick="<?php echo $onClick; ?>">
```
Comment (Quy) [inline administrator/templates/isis/html/layouts/joomla/pagination/link.php]: Add space after `$aria;`

**Comment by Quy** [comment]
See output.  `Page Previous` shouldn't be `aria-current="true"` right?? Can you have more than one `aria-current="true"`?
```
			<nav role="navigation" aria-label="Pagination">
			<ul class="pagination-list">
					<li class="disabled">
		<span aria-current="true" aria-label="Page Start (Page 1 of 4)">
			<span class="icon-backward icon-first" aria-hidden="true"></span>		</span>
	</li>
	<li class="disabled">
		<span aria-current="true" aria-label="Page Previous">
			<span class="icon-step-backward icon-previous" aria-hidden="true"></span>		</span>
	</li>
										<li class="active">
		<span aria-current="true" aria-label="Page 1">
			1		</span>
	</li>
```

**Comment by brianteeman** [comment]
no that looks wrong - i will check it again later

**Comment by brianteeman** [comment]
@quy open to suggestions/help - the code here is a bit funky

File: administrator/templates/isis/html/layouts/joomla/pagination/link.php [hunk 3]
```diff
@@ -95,12 +100,14 @@
...
 	<li<?php echo $liClass ? ' class="' . $liClass . '"' : ''; ?>>
-		<a <?php echo $cssClasses ? 'class="' . implode(' ', $cssClasses) . '"' : ''; ?> <?php echo $title; ?> href="#" onclick="<?php echo $onClick; ?>">
+		<a aria-label="<?php echo $aria ;?>" <?php echo $cssClasses ? 'class="' . implode(' ', $cssClasses) . '"' : ''; ?> <?php echo $title; ?> href="#" onclick="<?php echo $onClick; ?>">
 			<?php echo $display; ?>
 		</a>
 	</li>
 <?php else : ?>
 	<li class="<?php echo $class; ?>">
-		<span><?php echo $display; ?></span>
+		<span aria-current="true" aria-label="<?php echo JText::sprintf('JLIB_HTML_PAGE_CURRENT', $item->text); ?>">
```
Comment (Quy) [inline administrator/templates/isis/html/layouts/joomla/pagination/link.php]: This should do the trick. Please test.
```
<span <?php echo $class == 'active' ? 'aria-current="true"' : '' ?> aria-label="<?php echo JText::sprintf('JLIB_HTML_PAGE_CURRENT', $item->text); ?>">
```

File: administrator/templates/isis/html/layouts/joomla/pagination/link.php [hunk 4]
```diff
@@ -95,12 +100,14 @@
 ?>
 <?php if ($displayData['active']) : ?>
 	<li<?php echo $liClass ? ' class="' . $liClass . '"' : ''; ?>>
-		<a <?php echo $cssClasses ? 'class="' . implode(' ', $cssClasses) . '"' : ''; ?> <?php echo $title; ?> href="#" onclick="<?php echo $onClick; ?>">
+		<a aria-label="<?php echo $aria ;?>" <?php echo $cssClasses ? 'class="' . implode(' ', $cssClasses) . '"' : ''; ?> <?php echo $title; ?> href="#" onclick="<?php echo $onClick; ?>">
```
Comment (Quy) [inline administrator/templates/isis/html/layouts/joomla/pagination/link.php]: Move the space after `;`

File: administrator/templates/isis/html/layouts/joomla/pagination/link.php (hunk 3 above)
Comment (brianteeman) [reply administrator/templates/isis/html/layouts/joomla/pagination/link.php]: thats similar to one of the things i tried - but it doesnt work as it produces
````
<li class="disabled">
		<span aria-label="JLIB_HTML_PAGE_CURRENT">
			<span class="icon-step-backward icon-previous" aria-hidden="true"></span>	
		</span>
</li>

Comment (brianteeman) [reply administrator/templates/isis/html/layouts/joomla/pagination/link.php]: ok should be fixed now - thanks

File: administrator/templates/isis/html/layouts/joomla/pagination/link.php [hunk 5]
```diff
@@ -95,12 +100,14 @@
...
 	<li<?php echo $liClass ? ' class="' . $liClass . '"' : ''; ?>>
-		<a <?php echo $cssClasses ? 'class="' . implode(' ', $cssClasses) . '"' : ''; ?> <?php echo $title; ?> href="#" onclick="<?php echo $onClick; ?>">
+		<a aria-label="<?php echo $aria; ?>" <?php echo $cssClasses ? 'class="' . implode(' ', $cssClasses) . '"' : ''; ?> <?php echo $title; ?> href="#" onclick="<?php echo $onClick; ?>">
 			<?php echo $display; ?>
 		</a>
 	</li>
 <?php else : ?>
 	<li class="<?php echo $class; ?>">
-		<span><?php echo $display; ?></span>
+	<span <?php echo $class == 'active' ? 'aria-current="true" aria-label="' . JText::sprintf('JLIB_HTML_PAGE_CURRENT', $item->text) . '"' : '' ?>>
```
Comment (Quy) [inline administrator/templates/isis/html/layouts/joomla/pagination/link.php]: Add semicolon after `: ''`

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 94bb83c99d9214a58ec330cee03df5cecd819861<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18357">issues.joomla.org/tracker/joomla-cms/18357</a>.</sub>

**Comment by ghost** [comment]
@Quy can you please explain how the Code looks without and with Pull Request?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18357">issues.joomla.org/tracker/joomla-cms/18357</a>.</sub>

**Comment by Quy** [comment]
Go to Content > Articles
View page source
Find `<ul class="pagination-list">`

Before PR:
pagination-before

After PR:
pagination

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 94bb83c99d9214a58ec330cee03df5cecd819861<br /><br />Thanks for Help, @Quy <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18357">issues.joomla.org/tracker/joomla-cms/18357</a>.</sub>

**Comment by ghost** [comment]
RTC after two successful tests.

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**